src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -319,6 +319,28 @@
       (middleware-utils/json-response 200 info-for-json))))
 
 (schema/defn ^:always-validate
+  all-tasks-response! :- ringutils/RingResponse
+  "Process the info, returning a Ring response to be propagated back up to the
+  caller of the endpoint.
+  Returns environment as a list of objects for an eventual future in which
+  tasks for all environments can be requested, and a given task will list all
+  the environments it is found in."
+  [info-from-jruby :- List
```

Comment:
Are you going with `List` for future compatibility? It seems like a slightly more specific schema, even if it was likely to change later, might be better.